lib/SPIRV/SPIRVRegularizeLLVM.cpp

```diff
@@ -335,6 +335,52 @@ bool SPIRVRegularizeLLVMBase::runRegularizeLLVM(Module &Module) {
   return true;
 }
 
+namespace {
+void regularizeWithOverflowInstrinsics(StringRef MangledName, CallInst *Call,
+                                       Module *M,
+                                       std::vector<Instruction *> &ToErase) {
+  IRBuilder Builder(Call);
+  Function *Builtin = Call->getModule()->getFunction(MangledName);
+  AllocaInst *A;
+  if (Builtin) {
+    StructType *StructBuiltinTy =
+        cast<StructType>(Builtin->getParamStructRetType(0));
+    A = Builder.CreateAlloca(Builtin->getParamStructRetType(0));
```

Comment:
(not-clang friendly suggesting)
```suggestion
    {
      IRBuilderBase::InsertPointGuard Guard(Builder);
      Builder.SetInsertPointPastAllocas(Call->getParent()->getParent());
      A = Builder.CreateAlloca(Builtin->getParamStructRetType(0));
    }
 ```
 Reason:
If the intrinsic appears in not-the-first basic block of the function the alloca will be created in this basic block, resulting in OpVariable created not-the-first basic block (and not-necessarily in the beginning of it). And this violates:
`All OpVariable instructions in a function must be in the first block in the function. These instructions, together with any intermixed OpLine and OpNoLine instructions, must be the first instructions in that block. (Note the validation rules prevent OpPhi instructions in the first block of a function.)`

I suggest to insert the alloca in the beginning of the function right away as it's cheap to do in the pass.
Same applicable to the create alloca below.

Reply:
Follows a notion from https://github.com/KhronosGroup/SPIRV-LLVM-Translator/pull/2439